rimage/depth_map_test.go

```diff
@@ -333,12 +333,53 @@ func TestDepthColorModel(t *testing.T) {
 	test.That(t, convGray.(color.Gray16).Y, test.ShouldEqual, 6168)
 }
 
-func TestDepthMapEncoding(t *testing.T) {
-	m, err := NewDepthMapFromFile(context.Background(), artifact.MustPath("rimage/fakeDM.vnd.viam.dep"))
+func TestViamDepthMap(t *testing.T) {
+	// create various types of depth representations
+	width := 10
+	height := 20
+	dm := NewEmptyDepthMap(width, height)
+	g16 := image.NewGray16(image.Rect(0, 0, width, height))
+	g8 := image.NewGray(image.Rect(0, 0, width, height))
+	for x := 0; x < width; x++ {
+		for y := 0; y < height; y++ {
+			dm.Set(x, y, Depth(x*y))
+			g16.SetGray16(x, y, color.Gray16{uint16(x * y)})
+			g8.SetGray(x, y, color.Gray{uint8(x * y)})
+		}
+	}
+	// write dm to a viam type buffer
+	buf := &bytes.Buffer{}
+	byt, err := WriteViamDepthMapTo(dm, buf)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, byt, test.ShouldEqual, (3*8 + 2*width*height)) // 3 bytes for header, 2 bytes per pixel
+	// write gray16 to a viam type buffer
+	buf16 := &bytes.Buffer{}
+	byt16, err := WriteViamDepthMapTo(g16, buf16)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, byt16, test.ShouldEqual, (3*8 + 2*width*height)) // 3 bytes for header, 2 bytes per pixel
+	// gray should fail
+	buf8 := &bytes.Buffer{}
+	_, err = WriteViamDepthMapTo(g8, buf8)
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "cannot convert image type")
+	// read from a viam type buffers and compare to original
+	dm2, err := ReadDepthMap(buf)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, dm2, test.ShouldResemble, dm)
+	dm3, err := ReadDepthMap(buf16)
 	test.That(t, err, test.ShouldBeNil)
+	test.That(t, dm3, test.ShouldResemble, dm)
+}
 
-	// Test values at points of DepthMap
-	// This example DepthMap (fakeDM) was made such that Depth(x,y) = x*y
+func TestDepthMapEncoding(t *testing.T) {
```

Comment:
I think we lost the "make sure you can read this from a file with the right extension" test and we might wanna keep it. The depth map in artifact/rimage/fakeDM.vnd.viam.dep is just like the one created below from scratch.

Reply:
This PR I think actually changes the way `vnd.viam.dep` files are written. Each value is not given 8 bytes of space in the file, but only two bytes. This "compression" makes the new file sizes smaller, but also means that the format is different from what it was before. 

But yes, I will create a new `vnd.viam.dep` file without the 8byte padding, and just the 2byte per pixel length.

Reply:
And here is the new vnd.viam.dep format, which is more compact
```
0000000   D   E   P   T   H   M   A   P 
0000010  024  \0  \0  \0  \0  \0  \0  \0
0000020  \n  \0  \0  \0  \0  \0  \0  \0  
0000030  \0  \0  \0  \0  \0  \0  \0  \0
0000040  \0  \0  \0  \0  \0  \0  \0  \0  
0000050  \0  \0  \0  \0  \0  \0  \0  \0
0000060  \0  \0  \0  \0  \0  \0  \0  \0  
0000070  \0  \0  \0  \0  \0  \0  \0  \0
0000100  \0  \0 001  \0 002  \0 003  \0 
0000110  004  \0 005  \0 006  \0  \a  \0
0000120  \b  \0  \t  \0  \n  \0  \v  \0  
0000130  \f  \0  \r  \0 016  \0 017  \0
0000140  020  \0 021  \0 022  \0 023  \0  
0000150  \0  \0 002  \0 004  \0 006  \0
```